nemo/collections/tts/modules/submodules.py

```diff
@@ -758,15 +758,11 @@ def forward(self, batch_size=None, speaker=None, reference_spec=None, reference_
             embs = self.lookup_module(speaker)
 
         # Get GST based speaker embedding
-        if self.gst_module is not None:
-            if reference_spec is None or reference_spec_lens is None:
-                raise ValueError(
-                    "You should add `reference_audio` in sup_data_types or remove `speaker_encoder`in config."
-                )
-            out = self.gst_module(reference_spec, reference_spec_lens)
-            embs = out if embs is None else embs + out
-
-        elif self.gst_module is None and reference_spec is not None and reference_spec_lens is not None:
-            logging.warning("You may add `gst_module` in speaker_encoder to use reference_audio.")
+        if reference_spec is not None and reference_spec_lens is not None:
+            if self.gst_module is not None:
+                out = self.gst_module(reference_spec, reference_spec_lens)
+                embs = out if embs is None else embs + out
+            else:
+                logging.warning("You may add `gst_module` in speaker_encoder to use reference_audio.")
```

Comment:
Why do we relax the logging level to `warning` from `errors`? I was wondering if we should raise errors when the behavior is not expected.

Reply:
We previously raise error when the model has `gst_module` and users don't provide `reference_spec`. 
However, there is a scenario we pre-compute the speaker embeddings of `gst_module`, so the model still has `gst_module` but we don't provide `reference_spec` anymore.

Reply:
Thanks. it is fine for me. @rlangman @redoctopus any thoughts?

Reply:
Fine with me, though could you add a sentence in the warning telling users what it does by default instead?